integration_test/cases/client_test.exs

```diff
@@ -127,26 +152,30 @@ defmodule ClientTest do
     try do
       P.run(pool, fn(conn) ->
         assert_receive :reconnected
-        spawn_link(fn() ->
```

Comment:
We want to spawn_link here so that checkout occurs before `run` or `execute` has chance to fail/checkin. I think this previous code should work with timeout of `100`.

Reply:
It didn't work with a higher timeout either. There is a race on which arguments are popped from the agent.